### ISSUE TITLE
Traffic Ops Server Unique Constraint Doesn't Allow Multiple Servers With The Same IP and Profile on Different Ports

### ISSUE THREAD

**Issue by rob05c** [apache/incubator-trafficcontrol#1762]
The Traffic Ops `server` table has a unique constraint on `(ip, profile)`. This should be `(ip, port)`, or possibly `(ip, port, profile)`, to allow multiple servers on the same machine/ip with different ports. This is particularly useful for testing with many synthetic servers on the same machine.

I particularly need this, because I was trying to create a Dockerfile with ~1000 test caches, to load test the Monitor, and it isn't possible without dropping that constraint (or creating 1000 separate profiles, which is infeasible).

*implementation*
The constraints are in `traffic_ops/app/db/create_tables.sql` 

```CREATE UNIQUE INDEX idx_89709_ip_profile ON server USING btree (ip_address, profile);```
```CREATE UNIQUE INDEX idx_89709_ip6_profile ON server USING btree (ip6_address, profile);```

A Goose migration needs created in `traffic_ops/app/db/migrations` which drops those indexes (constraints) and creates new ones for `(ip, port)`.

**Comment by mitchell852** [comment]
@rob05c - is this still relevant?

**Comment by rob05c** [comment]
Yes. This is still relevant, and still a pain for people running multiple CDN services on the same host.
